### PR TITLE
Column data model (2 - Bar Series)

### PR DESCRIPTION
**Pull request by jacobp100**

### CODE REVIEW

File: packages/ag-charts-community/src/chart/axis/__image_snapshots__/grouped-category-axis-test-ts-grouped-category-axis-examples-for-integrated-charts-grouped-category-axis-example-flip-it-should-render-to-canvas-as-expected-1-snap.png
Comment (jacobp100) [inline packages/ag-charts-community/src/chart/axis/__image_snapshots__/grouped-category-axis-test-ts-grouped-category-axis-examples-for-integrated-charts-grouped-category-axis-example-flip-it-should-render-to-canvas-as-expected-1-snap.png:1]: https://plnkr.co/edit/hMmfzJhk7I5chjeV?open=main.js

File: packages/ag-charts-community/src/chart/axis/groupedCategoryAxis.test.ts [hunk 1]
```diff
@@ -178,7 +178,12 @@ describe('Grouped Category Axis Examples', () => {
             };
 
             chart = await createChart(example.options);
-            await axisCompare();
+            try {
+                await axisCompare();
+            } catch (e) {
+                console.log(chart.getOptions());
+                throw e;
+            }
```
Comment (alantreadway) [inline packages/ag-charts-community/src/chart/axis/groupedCategoryAxis.test.ts:186]: Nit: still needed?

File: packages/ag-charts-community/src/chart/data/dataController.ts [hunk 2]
```diff
@@ -202,6 +202,7 @@ export class DataController {
                 dataModel,
                 processedData: {
                     ...processedData,
+                    // @ts-ignore
```
Comment (alantreadway) [inline packages/ag-charts-community/src/chart/data/dataController.ts:205]: Let's make sure to clean this out before final merge.

File: packages/ag-charts-community/src/chart/data/dataModel.ts [hunk 3]
```diff
@@ -828,6 +885,28 @@ export class DataModel<
...
+                    if (!adjustFn) continue;
+
+                    for (const index of group.indices) {
+                        adjustFn(undefined!, valueIndexes, index, columns);
+                    }
+                }
+
+                // for (const index of group.indices) {
+                //     updateDomains(values);
+                // }
```
Comment (alantreadway) [inline packages/ag-charts-community/src/chart/data/dataModel.ts:906]: Nit: Needed?

File: packages/ag-charts-community/src/chart/data/dataController.ts (hunk 2 above)
Comment (jacobp100) [reply packages/ag-charts-community/src/chart/data/dataController.ts:205]: Nope - will fix in PR 3

File: packages/ag-charts-community/src/chart/data/dataModel.ts (hunk 3 above)
Comment (jacobp100) [reply packages/ag-charts-community/src/chart/data/dataModel.ts:906]: Eventually. The bar series doesn't yet work if you turn off the row model - but if you do turn it off, you'll need something like this to update the domains

File: packages/ag-charts-community/src/chart/axis/groupedCategoryAxis.test.ts (hunk 1 above)
Comment (jacobp100) [reply packages/ag-charts-community/src/chart/axis/groupedCategoryAxis.test.ts:186]: Nope - will fix in PR 3

File: packages/ag-charts-community/src/chart/data/dataController.ts (hunk 2 above)
Comment (jacobp100) [reply packages/ag-charts-community/src/chart/data/dataController.ts:205]: Will do

**Pull request closed without merging**